numpy/core/src/npysort/quicksort.c.src

```diff
@@ -107,6 +111,7 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
                 *sptr++ = pi - 1;
                 pl = pi + 1;
             }
+            *psdepth++ = cdepth + 1;
```

Comment:
Not quite, it is semantically equivalent to
```
cdepth += 1;
*psdepth++ = cdepth;
``
i.e. cdepth itself is increased for the next pass through. There is no worry about actual stack depth, it is never larger than log2(n) because the largest partition gets pushed, so the loop will terminate in at most log2(n) passes, where n is the size on entry to the loop.

Thinking about this is making my head hurt ;)